src/net.h

```diff
@@ -642,13 +641,15 @@ class TransportDeserializer {
     // read and deserialize data
     virtual int Read(const char *data, unsigned int bytes) = 0;
     // decomposes a message from the context
-    virtual CNetMessage GetMessage(const CMessageHeader::MessageStartChars& message_start, int64_t time) = 0;
+    virtual Optional<CNetMessage> GetMessage(int64_t time) = 0;
     virtual ~TransportDeserializer() {}
 };
 
 class V1TransportDeserializer final : public TransportDeserializer
 {
 private:
+    const CChainParams& m_chain_params;
+    const NodeId node_id; // Only for logging
```

Comment:
nit: name `m_node_id` to comply with [style guidelines](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#coding-style-c)

Reply:
One day I'll remember, thanks!